app/main.py

```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_people = [Person(person["name"], person["age"]) for person in people]
+
+    for i, person in enumerate(people):
+        if person.get("wife"):
+            new_people[i].wife = Person.people[person["wife"]]
+        if person.get("husband"):
+            new_people[i].husband = Person.people[person["husband"]]
```

Comment:
It's not a good practice to use enumerate here because you cannot handle case when order of cache will change for some reason.